tests/resource_tests.cc

```diff
@@ -72,17 +76,29 @@ TEST_F(ResourceTest, MaterialAbsorbGraphid) {
 
 TEST_F(ResourceTest, MaterialExtractTrackid) {
   int obj_id = m1->obj_id();
-  Material::Ptr m3 = m1->ExtractQty(2);
+  Material::Ptr m4 = m1->ExtractQty(2);
```

Comment:
I wonder if we should rename all the local objects to avoid potential name collisions in the future
```suggestion
  Material::Ptr lm1 = m1->ExtractQty(2);
```
Add 'l' for "local" and restart numbering at 1?